include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -654,12 +654,46 @@ __group_scan_fits_in_slm(const sycl::queue& __q, std::size_t __n, std::size_t __
     return (__n <= __single_group_upper_limit && __max_slm_size >= __req_slm_size);
 }
 
+//This is a temporary data structure which is used to store results to registers during a reduce then scan operation.
+template <std::uint16_t elements, typename _ValueT>
+struct __temp_data_array
```

Comment:
At this point you describes `elements` as `std::uint16_t`
But in usages you are using `std::int32_t` type:
```C++
// balanced path
template <typename _CustomName, typename _Range1, typename _Range2, typename _Range3, typename _Compare,
          typename _SetTag>
__future<sycl::event, __result_and_scratch_storage<oneapi::dpl::__internal::__difference_t<_Range3>>>
__parallel_set_reduce_then_scan(oneapi::dpl::__internal::__device_backend_tag __backend_tag, sycl::queue& __q,
                                _Range1&& __rng1, _Range2&& __rng2, _Range3&& __result, _Compare __comp, _SetTag)
{
    constexpr std::int32_t __diagonal_spacing = 32;

    // ...
    using _TempData = __temp_data_array<__diagonal_spacing, _OutValueT>;
```

Probably better to fix this moment.

Reply:
changed the type of diagonal spacing to `uint16_t`,  thanks.